security/user-last-seen.php

```diff
@@ -0,0 +1,148 @@
+<?php
+namespace Automattic\VIP\Security;
+
+const LAST_SEEN_META_KEY = 'wpvip_last_seen';
+const LAST_SEEN_UPDATE_USER_META_CACHE_TTL = MINUTE_IN_SECONDS * 5; // Store last seen once every five minute to avoid too many write DB operations
+const LAST_SEEN_UPDATE_USER_META_CACHE_KEY_PREFIX = 'wpvip_last_seen_update_user_meta_cache_key';
+const GROUP = 'wpvip';
+
+add_filter( 'determine_current_user', function ( $user_id ) {
+	if ( ! $user_id ) {
+		return $user_id;
+	}
+
+	$cache_key = LAST_SEEN_UPDATE_USER_META_CACHE_KEY_PREFIX . $user_id;
+
+	if ( wp_cache_get( $cache_key, GROUP ) ) {
+		// Last seen meta was checked recently
+		return $user_id;
+	}
+
+	$is_api_request = ( ( defined( 'XMLRPC_REQUEST' ) && XMLRPC_REQUEST ) || ( defined( 'REST_REQUEST' ) && REST_REQUEST ) );
```

Comment:
If we add `use Automattic\VIP\Utils\Context` to the top, we can do this:

```suggestion
	$is_api_request = Context::is_rest_api() || Context::is_xmlrpc_api();
```